### PR TITLE
Allow Appmenu custom positioning

### PR DESCRIPTION
**Pull request by alimirjamali**
fixes: https://github.com/QubesOS/qubes-issues/issues/9093

### CODE REVIEW

**Comment by alimirjamali** [comment]
Pylint errors fixes, Most unittests updated, rpm_spec updated

**Comment by alimirjamali** [comment]
@marmarta this is ready for review

The patch for Appmenu positioning, Appmenu Settings patch and unittests is done.

File: qubes_menu/tests/test_appmenu.py [hunk 1]
```diff
@@ -86,3 +91,30 @@ def test_appmenu_options():
...
+    app_menu.appmenu_position = "mouse"
+    app_menu.reposition()
+    app_menu.appmenu_position = "top-left"
+    app_menu.reposition()
+    app_menu.appmenu_position = "top-right"
+    app_menu.reposition()
+    app_menu.appmenu_position = "bottom-left"
+    app_menu.reposition()
+    app_menu.appmenu_position = "bottom-right"
+    app_menu.reposition()
```
Comment (marmarta) [inline qubes_menu/tests/test_appmenu.py:120]: can we add checking here that reposition did actually move the menu?

**Review by marmarta — CHANGES_REQUESTED** [review]
I did not know how much I needed it, wow! I installed this PR for testing and I already love it. Thank you! A small change request on the test, but otherwise, love this.

**Comment by alimirjamali** [comment]
> I did not know how much I needed it, wow! I installed this PR for testing and I already love it. Thank you! A small change request on the test, but otherwise, love this.

Glad that you liked it. The unittest is now updated

**Review by marmarta — APPROVED** [review]

**Comment by marmarta** [comment]
@marmarek , this may also be helpful to combat the weird KDE bugs with location of the menu? Not sure about that, it is some wayland blood magic that, but maybe?

File: icons/qappmenu-top-right.svg [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
<rect width="24" height="24" transform="rotate(180 12 12)" opacity="0"/>
+
<path d="M18 7.05a1 1 0 0 0-1-1L9 6a1 1 0 0 0 0 2h5.56l-8.27 8.29a1 1 0 0 0 0 1.42 1 1 0 0 0 1.42 0L16 9.42V15a1 1 0 0 0 1 1 1 1 0 0 0 1-1z"/>
+
</g>
+
</g>
+
</svg>
```
Comment (DemiMarie) [inline icons/qappmenu-top-right.svg:10]: Nit: Please use LF instead of CRLF and add a trailing newline.

File: icons/qappmenu-top-left.svg [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
<rect width="24" height="24" transform="rotate(90 12 12)" opacity="0"/>
+
<path d="M17.71 16.29L9.42 8H15a1 1 0 0 0 0-2H7.05a1 1 0 0 0-1 1v8a1 1 0 0 0 1 1H7a1 1 0 0 0 1-1V9.45l8.26 8.26a1 1 0 0 0 1.42 0 1 1 0 0 0 .03-1.42z"/>
+
</g>
+
</g>
+
</svg>
```
Comment (DemiMarie) [inline icons/qappmenu-top-left.svg:10]: Nit: Please use LF instead of CRLF and add a trailing newline.

File: icons/qappmenu-bottom-right.svg [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+
<rect width="24" height="24" transform="rotate(-90 12 12)" opacity="0"/>
+
<path d="M17 8a1 1 0 0 0-1 1v5.59l-8.29-8.3a1 1 0 0 0-1.42 1.42l8.3 8.29H9a1 1 0 0 0 0 2h8a1 1 0 0 0 1-1V9a1 1 0 0 0-1-1z"/>
+
</g>
+
</g>
+
</svg>
```
Comment (DemiMarie) [inline icons/qappmenu-bottom-right.svg:10]: Nit: Please use LF instead of CRLF and add a trailing newline.

File: icons/qappmenu-bottom-left.svg [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
<rect width="24" height="24" opacity="0"/>
+
<path d="M17.71 6.29a1 1 0 0 0-1.42 0L8 14.59V9a1 1 0 0 0-2 0v8a1 1 0 0 0 1 1h8a1 1 0 0 0 0-2H9.41l8.3-8.29a1 1 0 0 0 0-1.42z"/>
+
</g>
+
</g>
+
</svg>
```
Comment (DemiMarie) [inline icons/qappmenu-bottom-left.svg:10]: Nit: Please use LF instead of CRLF and add a trailing newline.

**Comment by DemiMarie** [comment]
@marmarek can we use these SVG files?  The license on the website of the generator appears incompatible.

**Comment by marmarek** [comment]
@Demi just stop. You have better things to do.

File: icons/qappmenu-top-right.svg (hunk 2 above)
Comment (alimirjamali) [reply icons/qappmenu-top-right.svg:10]: Sure. I will do it. The originals SVGs (from Eva Icons under MIT license) were like this. Will change them accordingly.

File: icons/qappmenu-top-left.svg (hunk 3 above)
Comment (alimirjamali) [reply icons/qappmenu-top-left.svg]: All arrow svgs are redone in Inkscape with CC0 Public Domain

**Pull request merged**